File: ow-wasm-action/examples/add.rs
#[cfg(feature = "wasm")]
ow_wasm_action::pass_json!(func);

#[cfg(not(feature = "wasm"))]
ow_wasm_action::json_args!(func);

fn func(json: serde_json::Value) -> Result<serde_json::Value, anyhow::Error> {
    let param1 = json
        .get("param1")
        .ok_or_else(|| anyhow::anyhow!("Expected param1 to be present"))?
        .as_i64()
        .ok_or_else(|| anyhow::anyhow!("Expected param1 to be an i64"))?;

    let param2 = json
        .get("param2")
        .ok_or_else(|| anyhow::anyhow!("Expected param2 to be present"))?
        .as_i64()
        .ok_or_else(|| anyhow::anyhow!("Expected param2 to be an i64"))?;

    Ok(serde_json::json!({ "result": param1 + param2 }))
}

File: ow-assemblyscript-action/assembly/add.ts
import { JSON } from "assemblyscript-json";

function handler(json: JSON.Obj): JSON.Obj | null {

  let param1 = json.getInteger("param1")
  let param2 = json.getInteger("param2")

  if (param1 != null && param2 != null) {

    let result = new JSON.Obj()
    result.set("result", param1.valueOf() + param2.valueOf())
    return result

  } else {
    console.error("parameters `param1` or `param2` were not present")
    return null
  }

}